src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/AbstractDocumentationCommentService.cs

```diff
@@ -139,6 +95,63 @@ public string GetBannerText(TDocumentationCommentTriviaSyntax documentationComme
             return text;
         }
 
+        private void HandleElement(TXmlElementSyntax summaryElement, StringBuilder sb)
+        {
+            foreach (var node in summaryElement.ChildNodes())
+            {
+                HandleNode(node, sb);
+            }
+        }
+
+        private void HandleNode(SyntaxNode node, StringBuilder sb)
+        {
+            if (node.HasLeadingTrivia)
+            {
+                // Collapse all trailing trivia to a single space.
+                AddSpaceIfNotAlreadyThere(sb);
+            }
+
+            if (node is TXmlTextSyntax xmlText)
+            {
+                var textTokens = GetTextTokens(xmlText);
+                AppendTextTokens(sb, textTokens);
+            }
+            else if (node is TXmlElementSyntax xmlElement)
+            {
+                HandleElement(xmlElement, sb);
+            }
+            else if (node is TXmlEmptyElementSyntax xmlEmpty)
+            {
+                foreach (var attribute in GetAttributes(xmlEmpty))
+                {
+                    switch (attribute)
+                    {
+                        case TXmlCrefAttributeSyntax xmlCref:
+                            AddSpaceIfNotAlreadyThere(sb);
+                            sb.Append(GetCref(xmlCref).ToString());
+                            break;
+                        case TXmlNameAttributeSyntax xmlName:
+                            AddSpaceIfNotAlreadyThere(sb);
+                            sb.Append(GetIdentifier(xmlName).Text);
+                            break;
+                        case TXmlTextAttributeSyntax xmlTextAttribute:
+                            AddSpaceIfNotAlreadyThere(sb);
+                            AppendTextTokens(sb, GetTextTokens(xmlTextAttribute));
+                            break;
+                        default:
+                            Debug.Assert(false, $"Unexpected XML syntax kind {attribute.RawKind}");
```

Comment:
I think this one is more to help us catch a case we missed, without so negatively affecting the user experience.